### ISSUE TITLE
Transition fork users to official Caprine after new version

### ISSUE THREAD

**Issue by lefterisgar** [sindresorhus/caprine#1875]
I'm not sure about how to transition users of lefterisgar/caprine to upstream when the new release comes out. Maybe create a transitional release on my fork and change the repo to sindresorhus/caprine? Is that good enough or is there any better way?

**Comment by dusansimic** [comment]
One way to be completely sure that all users would switch to upstream version of Caprine is to create a release that would show a popup on startup saying they should switch to this version. It's a bit intrusive but it's the only way I can think of by which we can be sure. Also, updating readme on the fork could link other people who stumble upon that repo to upstream one.

I'm not sure weather auto updating could work. For rpm repos yes since a new release will just overwrite the fork but for anything else, users would probably need to switch manually.

**Comment by lefterisgar** [comment]
At least I know that Windows users will receive the update automatically and transition to sindresorhus/caprine without even noticing. AppImage updates are not yet enabled in the fork so Linux users (AppImage, deb, rpm) will need to update manually. I will no longer release Linux builds and both the release and readme will state what you've said.

**Comment by dusansimic** [comment]
> Unfortunately that's not the case on my fork. It doesn't have any repo, just GitHub releases.

That's my point, users who installed rpm packages had to install them manually (via command line). Since dnf doesn't yet differentiate pacakges from different repos once a new version is released to my copr repo it will overwrite the fork version once updated. With that in mind, things we need to worry about are AppImage and deb.

Now that I think of it. We could open an issue describing the migration process (for those who have automated updates, they will get the upstream version automatically; for those that have rpm packages just need to make sure they have the copr repo added; everyone else would need to manually install an update). That issue could be linked in a modal dialog that pops up in the forked version. What are your thoughts on this?

**Comment by lefterisgar** [comment]
>That's my point, users who installed rpm packages had to install them manually (via command line). Since dnf doesn't yet differentiate pacakges from different repos once a new version is released to my copr repo it will overwrite the fork version once updated. With that in mind, things we need to worry about are AppImage and deb.

Oh, I see! It took me a bit to get it! Just override the fork and it's done assuming users have the copr repo enabled.

>We could open an issue describing the migration process

Exactly!

>everyone else would need to manually install an update

Don't forget that Windows users will automatically receive the update too!

>That issue could be linked in a modal dialog that pops up in the forked version. What are your thoughts on this?

I might be missing something but what's the point? Either someone receives the update and is migrated to upstream or goes to GitHub releases and finds out the new release doesn't have any Linux build with a bold warning stating the reason for that. Isn't that easier than having to create a popup and mess with users' config files (so it only opens once)?

**Comment by lefterisgar** [comment]
Turns out the latest version of my fork was bugged (due to Electron 19) and the transition of 2.55.7-2 -> 2.55.7-3 won't work. However this doesn't affect this repo because we are now using Electron 20 (I have tested 2.55.7-3 -> 2.56.0 and it does work).

**Comment by sgtcoder** [comment]
I agree, there is something with the latest release. Features were stripped out, a lot of things broken, notifications broken, freezes up, high load.

**Comment by lefterisgar** [comment]
> Features were stripped out

Do you prefer features that don't work but still exist? We have hidden them until someone fixes them so users don't complain that they don't work.

> notifications broken

It is not specific to that release AFAIK

> freezes up, high load.

We are working on it.

**Comment by sgtcoder** [comment]
> Do you prefer features that don't work but still exist? We have hidden them until someone fixes them so users don't complain that they don't work.

It was mainly the ability to disable autoplay of the videos. If the window is open and a video is playing, it prevents the computer from going to suspend.

> freezes up, high load.
> We are working on it.

I just saw an update pushed out about 30 minutes ago, so I am going to test that out.

**Comment by lefterisgar** [comment]
> It was mainly the ability to disable autoplay of the videos. If the window is open and a video is playing, it prevents the computer from going to suspend.

We have recently received a report that it doesn't work anymore (#1845). After testing on my machine, I confirmed it. Did it still work for you?

> I just saw an update pushed out about 30 minutes ago, so I am going to test that out.

I hope you see a (small) difference! 😄

**Comment by sgtcoder** [comment]
Thank you. Seems a little better? It still hung up on me when getting messages, resulting in a high load and temp. My guess is the electron issue I believe I read about.

Also, just discovered you have your own repo! Thank you. Prior to that I was scraping from the GitHub API and pulling in my own repo server.
